satellite/marketing/marketingweb/server.go

```diff
@@ -9,31 +9,37 @@ import (
 	"net"
 	"net/http"
 	"path/filepath"
+	"reflect"
+	"time"
 
 	"github.com/gorilla/mux"
+	"github.com/gorilla/schema"
 	"github.com/zeebo/errs"
 	"go.uber.org/zap"
 	"golang.org/x/sync/errgroup"
+
+	"storj.io/storj/satellite/marketing"
 )
 
-// Error is satellite marketing error type
-var Error = errs.Class("satellite marketing error")
+var (
+	// Error is satellite marketing error type
+	Error   = errs.Class("satellite marketing error")
+	decoder = schema.NewDecoder()
+)
 
-// Config contains configuration for marketing offersweb server
+// Config contains configuration for marketingweb server
 type Config struct {
 	Address   string `help:"server address of the marketing Admin GUI" default:"127.0.0.1:8090"`
 	StaticDir string `help:"path to static resources" default:""`
 }
 
 // Server represents marketing offersweb server
 type Server struct {
-	log *zap.Logger
-
-	config Config
-
-	listener net.Listener
-	server   http.Server
-
+	log         *zap.Logger
+	Config      Config
```

Comment:
why is this public?

Reply:
When you told me to not use a fixed address earlier in my unit test I made this public so I could use the address string from the server config in my unit test.

Reply:
Maybe try adding this in the /marketingweb/server.go` ?
```golang
func (s *Server) Addr() string { return s.config.Address }
```

Reply:
I can do that but it's not clear to my why I would create a new method for the purpose of returning one of it's receiver's fields when I can just export the field to begin with ? 

Why is it bad to export Config?

Reply:
There's no reason to export it. The `s.config.Address` doesn't contain the actual address, but the template for the address... e.g. it could be `":0"`

Reply:
If do an actual request against that address it should fail.

Reply:
Hey @fuskovic, maybe try this?
```golang
s := planet.Satellites[0].Marketing.Listener.Addr()
resp, err := http.PostForm("http://"+s.String()+"/create", nil)
```